bin/bb8/remote_file_manager.py

```diff
@@ -0,0 +1,46 @@
+import json
+
+import docker
+from shellescape import quote
+
+from .remote_paths import RemotePaths
+
+
+class RemoteFileManager(object):
+    def __init__(self, paths: RemotePaths, docker_client=docker.from_env()):
+        self.paths = paths
+        self.docker_client = docker_client
+
+    def run_remote_cmd(self, remote_cmd):
+        cmd = ["ssh", self.paths.host, remote_cmd]
+        volumes = {
+            "bb8_ssh": {"bind": "/root/.ssh", "mode": "ro"}
+        }
+        return self.docker_client.containers.run("instrumentisto/rsync-ssh",
+                                                 command=cmd,
+                                                 volumes=volumes,
+                                                 remove=True)
+
+    def get_metadata(self):
+        remote_cmd = 'cat {path} || :'.format(path=self.paths.metadata_file)
+        output = self.run_remote_cmd(remote_cmd)
+        if output:
+            return json.loads(output)
+        else:
+            return None
+
+    def write_metadata(self, metadata):
+        json_content = json.dumps(metadata)
+        path = self.paths.metadata_file
+        cmd = "echo {data} > {path}".format(data=quote(json_content), path=path)
```

Comment:
Will we ever run into command length issues here? If so we can use python's [`tarfile`](https://docs.python.org/3/library/tarfile.html) to create a tar file on the fly and then `.container.put_archive()` (I think) for the equivalent of `docker cp`

Reply:
This is running on the remote machine, so those options might be tricky.

Reply:
ah, of course